### DISCUSSION TITLE
Missing OpenID Connect settings in custom OIDC-based identity provider (Keycloak 21)

### DISCUSSION THREAD

**Discussion by Kagemusha-san** [keycloak/keycloak#21150] [Archive]
We have a custom OIDC-based identity provider (extending OIDCIdentityProviderConfig and OIDCIdentityProvider classes) for one of our customers. We had to add two custom fields to the identity provider by copying some theme-resource files and adapting them. This, of course, only worked with the old admin UI. Since Keycloak 21 removed the old UI we have a workaround to provide additional data via environment variables.

The problem we're facing now, is that we're not even able to see the OpenID Connect settings in our custom provider, which prevents us from configuring and therefore using it at all.

![image](https://github.com/keycloak/keycloak/assets/8342530/1f45d850-d7a1-418e-8c13-da4f225201af)

(How) Is it possible to have a OIDC-based custom identity provider which behaves and looks the same as the default OIDC identity provider?

We are aware that the old template file won't work with the new React frontend, so maybe we have to restructure our identity provider. All we're looking for is having a custom OIDC-based identity provider to add some logic but that otherwise behaves/looks like the standard OIDC identity provider.

**Comment by dasniko** [comment]
see #15344

**Reply by Kagemusha-san** [reply]
Thanks for the link. So maybe this will be fixed in Keycloak 22 but it could also be Keycloak 23. What are the planned release dates for these versions?

**Reply by dasniko** [reply]
There are no planned dates. It will be released in the next major version, after it is finished.
Red Hat never tells you any dates. 🤷‍♂️

**Reply by Kagemusha-san** [reply]
That's unfortunate. I was just thinking there is a release schedule because last major releases came every 3-4 month.
